search/processor.cpp

```diff
@@ -164,7 +125,8 @@ Processor::Processor(Index & index, CategoriesHolder const & categories,
       {StringUtf8Multilang::kInternationalCode, StringUtf8Multilang::kEnglishCode},
       {StringUtf8Multilang::kDefaultCode}};
 
-  m_keywordsScorer.SetLanguages(langPriorities);
+  m_ranker.m_keywordsScorer.SetLanguages(langPriorities);
+  m_preRanker.m_ranker = &m_ranker;
```

Comment:
If you still need to create a reference to m_ranker in m_preRanker, why not to inverse dependency? From my point of view, m_preRanker should know about m_ranker, and call it on each PreResult1.